src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2852,30 +2860,27 @@ private static void ThrowInvalidField(string f)
             result.NodeTag = DocumentDatabase.ServerStore.NodeTag;
         }
 
-        private QueryDoneRunning MarkQueryAsRunning(IIndexQuery query, OperationCancelToken token)
+        private IndexQueryDoneRunning MarkQueryAsRunning(IIndexQuery query)
         {
-            var queryStartTime = DateTime.UtcNow;
-            var queryId = Interlocked.Increment(ref _numberOfQueries);
-
-            if (queryId == 1 && _didWork == false)
+            if (_firstQuery && _didWork == false)
```

Comment:
I assume that we don't need to worry about concurrent queries here?

Reply:
This was done to set a timeout on first batch of newly created index (test RavenDB_8847). I've assumed that even when we have more then 1 concurrent query, then it is still an acceptable behavior.

Reply:
ok